pkg/config/config.go

```diff
@@ -114,7 +127,13 @@ const (
 	// No config needed for x509 signer
 
 	// KMS
-	kmsSignerKMSRef = "signers.kms.kmsref"
+	kmsSignerKMSRef      = "signers.kms.kmsref"
+	kmsAuthAddress       = "signers.kms.auth.address"
```

Comment:
I'm not super familiar with this, so this might actually be worse. WDYT of renaming some of these configs so that it's clear which kms they refer to?

```
signers.kms.auth.address > signers.kms.hashivault.address
signers.kms.auth.path > signers.kms.hashivault.path
signers.kms.auth.role > signers.kms.hashivault.role
signers.kms.auth.spire.sock > signers.kms.spire.sock
signers.kms.auth.spire.audience > signers.kms.spire.audience
```

Reply:
My original thought was to use generic names so other kms solutions could potentially reuse them. I'm happy to rename, and since we're calling it experimental, we're not tied to any of these choices.

Reply:
I ended up with the following to align with the upstream sigstore values (sigstore pulls from these fields based on the KMS used that it discovers from the `kmsref`):

- `signers.kms.auth.address`: replaces the need to define the VAULT_ADDR environment variable
- `signers.kms.auth.token`: new, I should have included this before, it's the VAULT_TOKEN when using vault
- `signers.kms.auth.oidc.path`: renamed to OIDC
- `signers.kms.auth.oidc.role`: renamed to OIDC
- `signers.kms.auth.spire.sock`: these last two felt better under auth because it's used to set the `auth.token`
- `signers.kms.auth.spire.audience`